packages/components/dialog/src/dialog.ts

```diff
@@ -14,50 +18,89 @@ import styles from './dialog.scss.js';
  * @slot default - Body content for the dialog
  * @slot footer - Footer content for the dialog
  * @slot header - Header content for the dialog
+ * @slot close - Closing button (placed in header) for the dialog
+ * @slot header-buttons - More space for buttons for the dialog's header
  * @slot title - The title of the dialog
+ * @slot subtitle - The subtitle of the dialog
  */
 export class Dialog extends ScopedElementsMixin(LitElement) {
   /** @private */
   static get scopedElements(): ScopedElementsMap {
     return {
-      'sl-button-bar': ButtonBar
+      'sl-button': Button,
+      'sl-button-bar': ButtonBar,
+      'sl-icon': Icon
     };
   }
 
   /** @private */
-  static override styles: CSSResultGroup = styles;
+  static override styles: CSSResultGroup = [breakpoints, styles];
 
+  /** @private */
   @query('dialog') dialog?: HTMLDialogElement;
 
   /** Disables the ability to close the dialog using the Escape key. */
   @property({ type: Boolean, attribute: 'disable-close' }) disableClose = false;
 
+  /** Determines whether closing button (default one) should be shown in the top right corner. */
+  @property({ type: Boolean, attribute: 'closing-button' }) closingButton?: boolean;
+
   /** The ARIA role of the dialog. */
   @property() override role: 'dialog' | 'alertdialog' = 'dialog';
 
+  /** Alignment of the action buttons in the dialog footer
+   * @type {'start' | 'end'} */
+  @property({ reflect: true, attribute: 'buttons-align' }) buttonsAlign: Extract<ButtonBarAlign, 'start' | 'end'> =
+    'end';
+
   override connectedCallback(): void {
     super.connectedCallback();
 
     this.inert = true;
+
+    this.dialog?.setAttribute('closing', 'false');
+  }
+
+  override disconnectedCallback(): void {
+    this.dialog?.removeEventListener('animationend', event => this.#handleAnimationEnd(event));
+
+    super.disconnectedCallback();
   }
 
   override render(): TemplateResult {
     return html`
       <dialog
         @cancel=${this.#onCancel}
         @click=${this.#onClick}
+        @keydown=${this.#onKeydown}
```

Comment:
Personally I try to keep things sort alphabetically: so `@keydown` *after* `@close` :)